tests/queries/0_stateless/02335_column_ttl_expired_column_optimization.sh

```diff
@@ -4,7 +4,10 @@ CUR_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
 # shellcheck source=../shell_config.sh
 . "$CUR_DIR"/../shell_config.sh
 
-$CLICKHOUSE_LOCAL --path "$CLICKHOUSE_TEST_UNIQUE_NAME" -nm -q "
+data_path="$CLICKHOUSE_TMP/$CLICKHOUSE_TEST_UNIQUE_NAME"
+trap 'rm -fr "${data_path:?}"' EXIT
```

Comment:
Let's avoid using this

Reply:
Why? With `trap` it will be cleaned up even I send TERM/INT to the process, w/o it won't.

Reply:
Because previous attempt to improve tests using `trap` was unsuccessful

Reply:
I don't have strong objections to not use `trap` here since there are other tests that are not that nice in cleanup, but the trend to not use `trap` at all - does not sounds right to me.

Remember that, that [bug in bash](https://gist.github.com/azat/affbda3f8c6b5c38648d4ab105777d88) is not that generic, and should not be a problem here.